src/types.ts

```diff
@@ -1,16 +1,160 @@
-import { Numbers } from "web3";
+import { Address, Numbers } from "web3";
 
-export interface NativeCurrencyData {
+export type NativeCurrencyData = {
   name: string;
   symbol: string;
   decimals: number;
-}
+};
 
-export interface AddEthereumChainRequest {
+/**
+ * Request to add a new chain to the user's wallet.
+ *
+ * See [EIP-3085](https://eips.ethereum.org/EIPS/eip-3085) for more details.
+ */
+export type AddEthereumChainRequest = {
   chainId: Numbers;
   blockExplorerUrls?: string[];
   chainName?: string;
   iconUrls?: string[];
   nativeCurrency?: NativeCurrencyData;
   rpcUrls?: string[];
-}
+};
+
+/**
+ * Request to switch to a new chain and register it with the user’s wallet if it isn’t already recognized.
+ *
+ * See [EIP-2015](https://eips.ethereum.org/EIPS/eip-2015) for more details.
+ */
+export type UpdateEthereumChainRequest = {
+  chainId: Numbers;
+  blockExplorerUrl?: string;
+  chainName?: string;
+  nativeCurrency?: NativeCurrencyData;
+  rpcUrls?: string[];
+};
+
+/**
+ * Request to switch the wallet’s currently active chain.
+ *
+ * See [EIP-3326](https://eips.ethereum.org/EIPS/eip-3326) for more details.
+ */
+export type SwitchEthereumChainRequest = {
+  chainId: Numbers;
+};
+
+/**
+ * Request to return a list of owned assets for the given address.
+ *
+ * See [EIP-2256](https://eips.ethereum.org/EIPS/eip-2256) for more details.
+ */
+export type GetOwnedAssetsRequest = {
+  /**
+   * Ethereum address that owns the assets.
+   */
+  address: Address;
+  options?: {
+    /**
+     * Chain ID respecting EIP-155.
+     */
+    chainId?: Numbers;
+    /**
+     * Maximum number of owned assets expected by the dApp to be returned.
+     */
+    limit?: number;
+    /**
+     * Array of asset interface identifiers such as ['ERC20', 'ERC721'].
+     */
+    types?: string[];
+    /**
+     * Human-readable text provided by the dApp, explaining the intended purpose of this request.
+     */
+    justification?: string;
+  };
+};
+
+/**
+ * A single asset owned by the wallet user.
+ *
+ * See [EIP-2256](https://eips.ethereum.org/EIPS/eip-2256) for more details.
+ */
+export type OwnedAsset = {
+  /**
+   * Ethereum checksummed address of the asset.
+   */
+  address: Address;
+  /**
+   * Identifier for the chain on which the assets are deployed.
+   */
+  chainId: Numbers;
+  /**
+   * Asset interface ERC identifier, e.g., ERC20.
+   * Optional - EIP-1820 could be used.
```

Comment:
What does this mean, exactly, that "EIP-1820 could be used"? If this property is omitted, how should the user interpret it?

Reply:
I think I copied it from the EIP spec, I have no idea what does it mean :) I will remove it.